cuML/CMakeLists.txt

```diff
@@ -143,6 +143,10 @@ target_link_libraries(ml_test
                       pthread
                       z)
 
+###################################################################################################
+# - build examples -------------------------------------------------------------------------
+
+add_subdirectory(examples)
```

Comment:
Could we make building the examples an optional part of the build? I can see a future where if we add many examples they probably won't be needed for development in most instances.

Reply:
I understand your reasoning. The rational I put this into the top level CMake is that during CI it is ensured that the examples are compiled so that a commit does not accidentally break them. If I make this optional what do I need do change to make sure that CI jobs are configured with this option enabled?

Reply:
it can be enabled by default, or we can just ask @mike-wendt or @raydouglass to consider it

Reply:
I added a configure option to disable compilation of examples: DISABLE_EXAMPLES.